lib/logstash/codecs/telemetry.rb

```diff
@@ -450,23 +615,152 @@ def decode(data)
               @logger.error("Unsupported xform", :xform => xform)
             end
 
-          when TSCodecTLVType::TS_CODEC_TLV_TYPE_COMPRESSOR_RESET
-            @zstream = Zlib::Inflate.new
+        when TSCodecTLVTypeV2::TS_CODEC_TLV_TYPE_V2_GPB_COMPACT
+          #####decode_compact(msg)
+          v, msg = msg.unpack("a#{l}a*")
+
+          if @data_compressed == 1
+            decompressed_unit = @zstream.inflate(v)
```

Comment:
Can you confirm you tested all combinations;
i.e. gpb + [compressed|uncompressed], gpbkv + [compressed|uncompressed], json [compressed|uncompressed]